### PR TITLE
chore: Create a registry base class

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:
This PR introduces a base class for the methods that the Registry should support, and leaving the other methods in the current Registry class as implementation details.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/registry.py [hunk 1]
```diff
@@ -145,7 +146,413 @@ def get_registry_store_class_from_scheme(registry_path: str):
         return get_registry_store_class_from_type(registry_store_type)
 
 
-class Registry:
+class BaseRegistry(abc.ABC):
+    # Entity operations
+    def apply_entity(self, entity: Entity, project: str, commit: bool = True):
```
Comment (felixwang9817) [inline sdk/python/feast/registry.py:151]: did you want to make these `@abstractmethods`?

File: sdk/python/feast/registry.py [hunk 2]
```diff
@@ -145,7 +146,413 @@ def get_registry_store_class_from_scheme(registry_path: str):
...
+    ) -> List[RequestFeatureView]:
+        """
+        Retrieve a list of request feature views from the registry
+
+        Args:
+            allow_cache: Allow returning feature views from the cached registry
+            project: Filter feature views based on project name
+
+        Returns:
+            List of feature views
```
Comment (kevjumba) [inline sdk/python/feast/registry.py:420]: request feature view

**Review by kevjumba — APPROVED** [review]
/approve

File: sdk/python/feast/registry.py (hunk 2 above)
Comment (achals) [reply sdk/python/feast/registry.py:420]: updated, was copy pasta from the Registry class.

**Pull request merged**